nuxeo-liveconnect-core/src/main/java/org/nuxeo/ecm/liveconnect/core/AbstractLiveConnectBlobProvider.java

```diff
@@ -203,7 +203,14 @@ public O getOAuth2Provider() {
 
     @Override
     public SimpleManagedBlob toBlob(LiveConnectFileInfo fileInfo) throws IOException {
-        LiveConnectFile file = getFile(fileInfo);
+        LiveConnectFile file;
+        try {
+            file = getFile(fileInfo);
+        } catch (IOException e) {
+            // we don't want to crash everything if the remote file cannot be accessed
+            log.error("Failed to access file: " + fileInfo, e);
+            file = new ErrorLiveConnectFile(fileInfo);
```

Comment:
In this class there is IOException, UncheckedIOException and now ErrorLiveConnectFile.  In which cases would we use each of these?

Reply:
I don't really understand the question... These are different things.